src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java

```diff
@@ -374,6 +374,17 @@ public static String getPathWithoutBucket(String path) {
 
     }
 
+    /**
+     * String -> Path. This *should* not be necessary (use Paths.get(URI.create(...)) instead) , but it currently is
+     * on Spark because using the fat, shaded jar breaks the registration of the GCS FilesystemProvider.
```

Comment:
How do the FileSystem providers get registered?  Is there a plugin loader that searches the class path for them or do they register statically on class loading?  Is it possible that there's a class loader issue where spark initializes the FileSystems before it's loaded the classes in the client jar?